mysql-test/suite/innodb/t/max_trx_no_recovery.test

```diff
@@ -0,0 +1,138 @@
+--source include/have_innodb.inc
+--source include/count_sessions.inc
+# The embedded server tests do not support restarting.
+--source include/not_embedded.inc
+--source include/have_debug.inc
+--source include/have_debug_sync.inc
+
+--let transactions_count=10
+--let perl_inc_file=$MYSQL_TMP_DIR/max_trx_no_recovery.inc
+
+--let $i=$transactions_count
+# We need several tables to avoid blocking on records locking and pages latching
+while($i) {
+--dec $i
+if ($i == 0) {
+--eval CREATE TABLE t_1_$i(a INT primary key) Engine=InnoDB STATS_PERSISTENT=0
+--eval INSERT INTO t_1_$i SET a = 10
+}
+--eval CREATE TABLE t_2_$i(a INT primary key) Engine=InnoDB STATS_PERSISTENT=0
+--eval INSERT INTO t_2_$i SET a = 10
+}
+--source include/wait_all_purged.inc
+
+--connect (prevent_purge,localhost,root,,)
+START TRANSACTION WITH CONSISTENT SNAPSHOT;
+
+# Let all the following redo logs will be pushed in the same rseg
+SET GLOBAL DEBUG_DBUG ="+d,assign_same_rseg";
```

Comment:
I was curious if this logic (of forcing rollback segment 0, which resides in the system tablespace) would be broken when the server is run with `innodb_undo_tablespaces=3`, but luckily that does not seem to be the case.